modules/ui/src/main/scala/lucuma/ui/forms/FormInputEV.scala

```diff
@@ -99,34 +106,87 @@ object FormInputEV {
       props.onValidChange(validated.isValid) >> cb(validated)
     }
 
+    def getInputElement(e: html.Element): CallbackOption[html.Input] = {
+      def childChain(ele: html.Element): Chain[html.Element] =
+        Chain.fromSeq(
+          (0 until ele.childElementCount).map(i => ele.children.item(i).asInstanceOf[html.Element])
+        )
+
+      def findWithClass(ele: html.Element, acc: Chain[html.Element]): Chain[html.Element] = {
+        val newAcc   = if (ele.className.contains("input")) acc :+ ele else acc
+        val children = childChain(ele)
+        if (children.isEmpty) newAcc
+        else newAcc ++ children.flatMap(c => findWithClass(c, Chain.empty))
+      }
+      // val oc = findWithClass(e, Chain.empty).headOption.map(_.firstChild.asInstanceOf[html.Input])
+      CallbackOption.liftOption(
+        findWithClass(e, Chain.empty).headOption.map(_.firstChild.asInstanceOf[html.Input])
```

Comment:
Nice optimization with `Chain` as accumulator.

Since we are only interested in `headOption`, can we cut the search short as soon as we find the first element?

Reply:
Actually, it seems the search for the `input` node is being done on each keystroke. Maybe it can be done once on `componentDidMount` and stored in the `Backend` or `State` ?

Reply:
Good idea for storing the input. I'll do that.

Reply:
I'm also going to do this in a separate PR in order to expedite releasing the functionality.